readthedocs/projects/views/private.py

```diff
@@ -264,6 +265,27 @@ def is_advanced(self):
         return data.get('advanced', True)
 
 
+@login_required
+def send_mail(request, project_slug):
+    """Sends abandoned project email."""
+    project = Project.objects.get(slug=project_slug)
+    proj_name = project_slug
+    for user in project.users.all():
+        if AdminPermission.is_admin(user, project):
+            email = user.email
```

Comment:
Can this be done on another way (not sure how we can _decide_ to which owner send the email)? Probably a `break` is needed here.

Reply:
Also I think this is insecure, since any logged in user can send a request here, right?

Reply:
I made this button to be visible only to a superuser in the html template. Does that look fine?

Reply:
Also, I have added a ``break`` statement. Should the mail be sent to a list of all users?

Reply:
Yes, but any user could do a manual request, so probably is worth to check for the superuser here. I'm not sure what is the best way of do it here anyway, probably the `required_login` decorator accepts a argument or there is a `super_user_only` decorator?

Maybe @humitos can help us here.

Reply:
https://stackoverflow.com/questions/12003736/django-login-required-decorator-for-a-superuser
@stsewd Sorry for the delay in reply. Will this work for us?

Reply:
That solution looks great! Let's wait to someone with more experience help us here, anyway, you could try testing this on your local instance and see if that works as expected.

Reply:
Sure @stsewd! Thanks 😄

Reply:
Sure @stsewd! Thanks 😄

Reply:
@stsewd I tested this on my local instance and it works as expected!

Reply:
@ericholscher Can you please guide me here!

Reply:
This looks like it is checking for project owners who are admins. We just want to send the email to _all_ project owners, but only let *admins* do it. We should be doing a `request.user.is_superuser` or similar check to determine if the logged in user is an admin, and then send the email to all project owners.

Reply:
The button for sending mail is only visible to the admins i.e. I have done a ``request.user.is_superuser`` in the html template. Also, we can check for superuser using a decorator like this: https://stackoverflow.com/questions/12003736/django-login-required-decorator-for-a-superuser. Does that look fine @ericholscher?